sql/parser/walk.go

```diff
@@ -23,14 +23,18 @@ import "fmt"
 // WalkExpr. The returned Expr replaces the pointer to the visited expression
 // in the parent node and can be used for rewriting expressions.
 type Visitor interface {
-	Visit(expr Expr) Expr
+	Visit(expr Expr, pre bool) (Visitor, Expr)
```

Comment:
Short for `preorder`, but that probably isn't the right terminology. Let me think if there is a better term.